app/views/search/history.html.erb

```diff
@@ -27,10 +27,84 @@
             </td>
             <td>
               <%= link_to "Execute Search", search_path(search), :class => "btn btn-primary btn-sm" %>
+              &nbsp;&nbsp;
+              <button id="btn-share-<%= search.id %>" data-serler-search-id="<%= search.id %>" class="btn btn-warning btn-sm" >Share</button>
             </td>
           </tr>
         <% end %>
+        <script>
+          $().ready(function(){
```

Comment:
Those javascript code lines should be in `app/assets/javascript/search.coffee`.
Don't write them in html file. It could be a good practice for you.